doc/usecases/kdb/UC_modify_config.md

```diff
@@ -0,0 +1,28 @@
+# Use Case: Modifying application configuration
+
+## Summary
+
+- **Title:** Modifying application configuration
+- **Scope:** `libelektra-kdb`
+- **Level:** Developer Goal
+- **Actors:** Application, Elektra
+- **Brief:** Application modifies configuration
+
+## Scenarios
+
+- **Precondition:** -
+- **Main success scenario:**
+  - [Application loads configuration](UC_load_config.md)
+  - Application modifies configuration in memory
+  - [Application stores configuration](UC_store_config.md)
+- **Alternative scenario:** -
+- **Error scenario:** -
+  > **Note**: The error scenarios from the references use cases still apply.
+  > However, the in-memory modification part itself has no error scenario.
+- **Postcondition:**
+  - The modified configuration MUST be stored into the KDB exactly.
```

Comment:
Here we come to the interesting discussions. E.g. this would influence what `spec` needs to do:

```suggestion
  - The modified configuration MUST be stored into the KDB exactly.
    Metadata from the specification gets added to the configuration.
```

@tmakar

Reply:
> Metadata from the specification gets added to the configuration.

What do you mean here?

1. `spec` should **remove** the metadata it added during `kdbSet`, so that it is **not stored** with the configuration
2. If specification is used at all here (which the use case intentionally doesn't mention because a specification is not required), then it would have to exist as a precondition.

Reply:
Still not sure what you mean. Remember this should describe the ideal case "how it should be" not "how it is". I know the current `spec` leaves some stuff around, but in principle `spec` should remove everything it generated.

In any case this is definitely out of scope here.